src/goto-instrument/contracts/assigns.cpp

```diff
@@ -133,8 +133,7 @@ assigns_clauset::conditional_address_ranget::generate_snapshot_instructions()
 
   instructions.add(goto_programt::make_assignment(
     upper_bound_address_var,
-    minus_exprt{plus_exprt{slice.first, slice.second},
-                from_integer(1, slice.second.type())},
+    plus_exprt{slice.first, slice.second},
```

Comment:
Why was the "-1" necessary before, or why is this the only change that's necessary?

Reply:
I guess we were expecting some "x <= y" checks to become "x < y" checks, so I too am curious to know why that didn't need to happen.

Reply:
I'd also be curious to know why changing the actual value used does not appear to break anything. Was there a bug before, or is there an edge case now not being tested?

Reply:
these expressions define upper bounds for intervals. We just interpret them as being included/excluded of the interval, but the inclusion check for two intervals just compares the upper bounds using `_ <= _`. Our test suite was missing a case with `malloc(0)`, and for `__CPROVER_max_malloc_size`. My only fear removing the `-1` would be a pointer overflow into the object bits, but it seems like the fact that pointers pointing to one past an object's boundary are legal in C is already factored in `__CPROVER_max_malloc_size`, so everything works.